packages/fast-components/src/design-system-consumer/design-system-consumer.ts

```diff
@@ -0,0 +1,76 @@
+import { observable } from "@microsoft/fast-element";
+import { DesignSystemResolverEntry } from "../styles/recipes";
+import { composedParent } from "../utilities";
+import { DesignSystemProvider } from "../design-system-provider";
+
+export interface ConsumerArgs {
+    recipes: DesignSystemResolverEntry[];
+}
+
+export interface DesignSystemConsumer {
+    recipes: DesignSystemResolverEntry[];
+    provider: DesignSystemProvider | null;
+}
+
+export function designSystemConsumer<T extends { new (...args: any[]) }>(
+    constructor: T
+): T;
+export function designSystemConsumer<T extends { new (...args: any[]) }>(
+    args: ConsumerArgs
+): <_T>(contructor: _T) => _T;
+export function designSystemConsumer<T extends { new (...args: any[]) }>(
+    argsOrConstructor: any
+): any {
+    function decorator(constructor: T, options: ConsumerArgs): T {
+        class Consumer extends constructor implements DesignSystemConsumer {
+            public readonly recipes = options.recipes;
+
+            @observable
+            public provider: DesignSystemProvider | null = null;
+
+            /**
+             * Find the parent DesignSystem provider.
+             */
+            public findProvider(): DesignSystemProvider | null {
+                let parent = composedParent(this as any);
+
+                while (parent !== null) {
+                    if ((parent as any).isDesignSystemProvider) {
```

Comment:
One way I think we could enhance this is to check also check if the parent is a *consumer* - if it is, then we can just attach to it's provider. I was also considering checking *siblings* if they are a consumer but thought that may start to get inefficient - thoughts?

Reply:
I don't think we should check siblings, but attaching to the parent makes sense to me. I wouldn't block on implementing that optimization though.